zerver/lib/actions.py

```diff
@@ -2023,13 +2023,23 @@ def do_add_default_stream(realm, stream_name):
     stream, _ = create_stream_if_needed(realm, stream_name)
     if not DefaultStream.objects.filter(realm=realm, stream=stream).exists():
         DefaultStream.objects.create(realm=realm, stream=stream)
+        event = dict(
+            type='update_default_streams',
+            default_streams=do_get_default_streams_for_realm(realm)
+        )
+        send_event(event, active_user_ids(realm))
 
 def do_remove_default_stream(realm, stream_name):
     # type: (Realm, text_type) -> None
     stream = get_stream(stream_name, realm)
     if stream is None:
         raise JsonableError("Stream does not exist")
     DefaultStream.objects.filter(realm=realm, stream=stream).delete()
+    event = dict(
+        type='update_default_streams',
+        default_streams=do_get_default_streams_for_realm(realm)
+        )
+    send_event(event, active_user_ids(realm))
```

Comment:
It might make sense to refactor this code into a `notify_default_streams` function shared by the add and remove cases